core/src/main/java/org/dcsa/conformance/core/traffic/ConformanceMessageBody.java

```diff
@@ -1,22 +1,45 @@
 package org.dcsa.conformance.core.traffic;
 
+import static org.dcsa.conformance.core.toolkit.JsonToolkit.OBJECT_MAPPER;
+
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
 import lombok.Getter;
 import lombok.ToString;
 
-import static org.dcsa.conformance.core.toolkit.JsonToolkit.OBJECT_MAPPER;
-
 @Getter
 @ToString
 public class ConformanceMessageBody {
-  private final boolean isCorrectJson;
+  private final boolean isRawByteArray;
+  private final byte[] byteArrayBody;
   private final String stringBody;
   private final JsonNode jsonBody;
+  private final boolean isCorrectJson;
+
+  public ConformanceMessageBody(byte[] byteArrayBody) {
+    this.isRawByteArray = true;
+    this.byteArrayBody = byteArrayBody == null ? new byte[0] : byteArrayBody;
+    this.stringBody = new String(this.byteArrayBody, StandardCharsets.UTF_8);
```

Comment:
How does this work if the content is not valid UTF-8 (which is likely for the use-cases that need raw byte transfers)?

Reply:
I specifically need support for https://app.swaggerhub.com/apis/dcsaorg/DCSA_EBL_PINT/3.0.0-Beta-1#/Transfer%20additional%20document/put_v3_envelopes__envelopeReference__additional_documents__documentChecksum_, which is about transferring additional documentation.

Said documentation can be **any** file including but not limited to PDF and will be transferred in the application/octet-stream stream. The validation will need to assert this stream matches the expected checksum, which is why we need binary transfers. However, there is no rule stating that the content will be a valid UTF-8 byte sequence. In fact, for a PDF, a zip file, scanned images, it will very likely **not** be valid UTF-8.